feat: build 7.3.4 on core24

wallyworld opened the pull request:
Update snapcraft yaml to build on core24.

Build latest mongodb 7.x and relevant tools binaries.

Tested on arm64 only

`snapcraft remote-build --launchpad-accept-public-upload --build-for=arm64`

hpidcock commented on snap/snapcraft.yaml:
```diff
@@ -72,22 +72,29 @@ parts:
       export GOPATH=$(realpath ..)
+      # mongo tools hard codes the build platform to bionic
+      # and panics if built on anything else :-(
+      # Tricking it into thinking it's running in CI
```
Magic.

hpidcock reviewed (approved):
Happy to approve so we can get CI jobs setup for this branch.

wallyworld committed:
feat: build 7.3.4 on core24
Update snapcraft yaml to buil on core24
Build latest mongodb 7.x and relevant tools binaries.

hpidcock committed:
fix: build mongo 7.3

hpidcock committed:
fix: add rust and cargo for ppc64el/s390x

hpidcock committed:
fix: add libgoogle-perftools4t64
